source/module_io/write_Vxc.hpp

```diff
@@ -0,0 +1,210 @@
+        const ModulePW::PW_Basis& rho_basis, const ModulePW::PW_Basis& rhod_basis, const ModuleBase::matrix& vloc,
+        const Charge& chg, LCAO_Hamilt& uhm, LCAO_Matrix& lm, Local_Orbital_Charge& loc,
+        const K_Vectors& kv, const ModuleBase::matrix& wg, Grid_Driver& gd)
+    {
+        ModuleBase::TITLE("ModuleIO", "write_Vxc");
+        const Parallel_Orbitals* pv = lm.ParaV;
+        int nbands = wg.nc;
+        // 1. real-space xc potential
+        // ModuleBase::matrix vr_xc(nspin, chg.nrxx);
+        double etxc = 0.0;
+        double vtxc = 0.0;
+        // elecstate::PotXC* potxc(&rho_basis, &etxc, vtxc, nullptr);
+        // potxc.cal_v_eff(&chg, &ucell, vr_xc);
+        elecstate::Potential* potxc = new elecstate::Potential(&rhod_basis, &rho_basis, &ucell, &vloc, &sf, &etxc, &vtxc);
+        std::vector<std::string> compnents_list = { "xc" };
+        potxc->pot_register(compnents_list);
+        potxc->update_from_charge(&chg, &ucell);
+
+        // 2. allocate AO-matrix 
+        // R (the number of hR: 1 for nspin=1, 4; 2 for nspin=2)
+        int nspin0 = (nspin == 2) ? 2 : 1;
+        std::vector<hamilt::HContainer<TR>> vxcs_R_ao(nspin0, hamilt::HContainer<TR>(pv));
+        for (int is = 0;is < nspin0;++is) vxcs_R_ao[is].set_zero();
+        // k (size for each k-point)
+        std::vector<TK> vxc_k_ao(pv->nloc);
+
+        // 3. allocate operators and contribute HR
+        // op (corresponding to hR)
+        typename TGint<TK>::type* gint = nullptr;
+        set_gint_pointer<TK>(uhm, gint);
+        std::vector<hamilt::Veff<hamilt::OperatorLCAO<TK, TR>>*> vxcs_op_ao(nspin0);
+        for (int is = 0;is < nspin0;++is)
+        {
+            vxcs_op_ao[is] = new hamilt::Veff<hamilt::OperatorLCAO<TK, TR>>(gint, &loc, &lm, kv.kvec_d, potxc, &vxcs_R_ao[is], &vxc_k_ao, &ucell, &gd, pv);
+            GlobalV::CURRENT_SPIN = is; //caution: Veff::contributeHR depends on GlobalV::CURRENT_SPIN
+            vxcs_op_ao[is]->contributeHR();
+        }
+#ifdef __EXX
+        hamilt::OperatorEXX<hamilt::OperatorLCAO<TK, TR>> vexx_op_ao(&lm, nullptr, &vxc_k_ao, kv);
+        // ======test=======
+        // std::vector<TK> test_vexxonly_k_ao(pv->nloc);
+        // hamilt::OperatorEXX<hamilt::OperatorLCAO<TK, TR>> test_vexxonly_op_ao(&lm, nullptr, &test_vexxonly_k_ao, kv);
+        // ======test=======
+#endif
+
+        //3. calculate and write the MO-matrix Exc
+        Parallel_2D p2d;
+        set_para2d_MO(*pv, nbands, p2d);
+
+        // ======test=======
+        // double total_energy = 0.0;
+        // double exx_energy = 0.0;
+        // ======test=======
+        for (int ik = 0;ik < kv.nks;++ik)
+        {
+            ModuleBase::GlobalFunc::ZEROS(vxc_k_ao.data(), pv->nloc);
+            int is = GlobalV::CURRENT_SPIN = kv.isk[ik];
+            dynamic_cast<hamilt::OperatorLCAO<TK, TR>*>(vxcs_op_ao[is])->contributeHk(ik);
+#ifdef __EXX
+            if (GlobalC::exx_info.info_global.cal_exx) vexx_op_ao.contributeHk(ik);
```

Comment:
EXX term is also writed, I suggest you describe this in annotation of this function and documentation.

Reply:
Another question is why don't you consider strong correlation term DFTU?

Reply:
> Another question is why don't you consider strong correlation term DFTU?

I will talk to @minyez about this. The problem is whether to output the well-defined XC part or the part that GW needs: If DFTU term is included, Should DeePKS term also be included?

Reply:
We decided to include the DFTU term but not include the DeePKS term (as noted in the document).